packages/sparql-language-server/src/SparqlLanguageServer.ts

```diff
@@ -88,6 +89,119 @@ export class SparqlLanguageServer extends AbstractLanguageServer<
     };
   }
 
+  parseDocument(document: TextDocument) {
+    const content = document.getText();
+    const { indices, unescapedString } = unescapeString(content);
+    const { cst, errors, ...otherParseData } = this.parser.parse(
+      unescapedString
+    );
+    const tokens = this.parser.input;
+
+    return {
+      cst: indices.length ? this.adjustCstForEscapedText(cst, indices) : cst,
+      tokens: indices.length
+        ? tokens.map((token) => this.adjustTokenForEscapedText(token, indices))
+        : tokens,
+      errors: indices.length
+        ? this.adjustErrorsForEscapedText(errors, indices)
+        : errors,
+      otherParseData: otherParseData as Omit<
+        ReturnType<StardogSparqlParser['parse']>,
+        'cst' | 'errors'
+      >,
+    };
+  }
+
+  adjustCstForEscapedText(input: Object | Object[], indices: number[]) {
```

Comment:
I don't believe any of the `adjust...` methods here and below refer to anything on `SparqlLanguageServer` other than calls to the other methods, so it may be better to define these within `stardog-language-utils` so they're also accessible outside `SparqlLanguageServer`.